### DISCUSSION TITLE
[Script request] FileFlows

### DISCUSSION THREAD

**Discussion by GCustom** [community-scripts/ProxmoxVE#324] [Request script]
### Application Name

Fileflows

### Website

https://fileflows.com/

### Description

FileFlows is a powerful, open-source tool for automating media file processing workflows, including encoding, decoding, and media management. It offers an intuitive GUI and extensive plugin support, making it ideal for tasks like video transcoding, organizing, and managing large media libraries.

Proposal:
Adding FileFlows to Proxmox VE Helper-Scripts would simplify installation and configuration for users who want to leverage their Proxmox environment for media processing. FileFlows’ integration with hardware acceleration (Intel Quick Sync Video, NVIDIA NVENC, etc.) allows users to optimize performance on capable hardware, making it an excellent tool for transcoding and other GPU-intensive tasks on Proxmox.

Benefits:
	1.	Simplified Setup: Proxmox users could benefit from a one-click installation script for FileFlows, reducing the complexity of installing dependencies like .NET SDK, FFmpeg, and GPU drivers.
	2.	Enhanced Media Management: FileFlows provides robust media management tools that are particularly useful for users running media servers on Proxmox, such as Plex, Jellyfin, or Emby.
	3.	Efficient Hardware Utilization: With proper setup, FileFlows can harness the power of Intel, NVIDIA, or AMD GPUs to accelerate video processing, allowing users to manage large media libraries without heavy CPU loads.
	4.	Community Demand: FileFlows has a growing user base in the media server community, and having it available on Proxmox with a helper script would meet a significant demand for easy access to media processing tools.

Example Script Functionality:
	•	Automated Installation: Downloads and installs FileFlows, FFmpeg with necessary codecs, and .NET dependencies.
	•	GPU Acceleration Configuration: Guides the user through enabling Intel Quick Sync, NVIDIA NVENC, or AMD hardware acceleration support.
	•	Path Configuration: Allows easy integration with existing media directories on Proxmox storage volumes.

This integration would make Proxmox an even more attractive option for users interested in building a self-hosted, hardware-accelerated media processing setup. By adding FileFlows to Proxmox VE Helper-Scripts, users could streamline their workflows for transcoding and managing media files directly within the Proxmox environment.

Thank you for your consideration, there is some overlap with Tdarr but the goal is different as it handles more types of flows, and some say the flows are easier to configure, that's been my experience so far.

Having this as a VM or LXC would be appreciated. VM can update it ourselves, LXC same but could also use your updater script or re-run this one to pull their new build down on top of the existing one.

### Due Diligence

- [X] I have searched existing [scripts](https://community-scripts.github.io/Proxmox/scripts) and found no duplicates.
- [X] I have searched existing [discussions](https://github.com/community-scripts/ProxmoxVE/discussions?discussions_q=) and found no duplicate requests.
- [X] This is not a game-related request.

**Comment by leah-potato** [comment]
Would love to see this as well.

**Comment by kalua85** [comment]
Take my vote for this, very nice app and a script to easily install would be great

**Comment by kkroboth** [comment]
I was interested in FileFlows after using Tdarr for some time. Found this thread and decided to start creating the script. I'm new to FileFlows and haven't full tested it, but I think I got the basics down. I'm going to debug more when I have time and PR it against their development repo.

* https://github.com/community-scripts/ProxmoxVED/compare/main...kkroboth:ProxmoxVED:lxc-fileflows?expand=1
* The update script needs to be implemented

I copied snippets from Tdarr and kept the same resource defaults.

>  • Path Configuration: Allows easy integration with existing media directories on Proxmox storage volumes.

Do you mean having options to add mount points in the LXC? I haven't seen that in any of the helper scripts, but that's an interesting idea!

**Comment by kkroboth** [comment]
Ready for testing! https://github.com/community-scripts/ProxmoxVED/issues/103

**Comment by Feramance** [comment]
Can the script be adjusted to add optional support for both AMD and NVidia GPUs too?

And what about having another script to install just a node for fileflows?

**Reply by djismgaming** [reply]
A fileflows node. That's what I'm looking for.
